pkg/bicep/bicep.go

```diff
@@ -23,19 +29,109 @@ var template = map[string]interface{}{
 }
 
 func GenerateFiles(buildPath, stepPath string, b *bundle.Bundle, fs afero.Fs) error {
-	p := path.Join(buildPath, stepPath, paramsFile)
+	p := path.Join(buildPath, stepPath, ParamsFile)
 	params, _ := inputvars.TranspileDevParams(p, b, fs)
 	connections, _ := inputvars.TranspileConnectionVarFile(p, b, fs)
 
-	template["parameters"].(map[string]interface{})["params"] = params
-	template["parameters"].(map[string]interface{})["md_metadata"] = inputvars.DefaultMdMetadata(b.Name)
-	template["parameters"].(map[string]interface{})["connections"] = connections
+	template["parameters"].(map[string]interface{})["params"] = map[string]any{"value": params}
+	template["parameters"].(map[string]interface{})["md_metadata"] = map[string]any{"value": inputvars.DefaultMdMetadata(b.Name)}
+	template["parameters"].(map[string]interface{})["connections"] = map[string]any{"value": connections}
 
 	content, err := json.MarshalIndent(template, "", "    ")
 
 	if err != nil {
 		return err
 	}
 
-	return afero.WriteFile(fs, fmt.Sprintf("%s/%s", stepPath, paramsFile), content, 0755)
+	return afero.WriteFile(fs, fmt.Sprintf("%s/%s", stepPath, ParamsFile), content, 0755)
+}
+
+// reconcileParams reads the params file keeping the md_metadata field intact as it's
+// not represented in the UI yet, adds the incoming params, and writes the file back out.
+func ReconcileParams(baseDir string, params map[string]any) error {
+	paramPath := path.Join(baseDir, ParamsFile)
+
+	fileParams := make(map[string]any)
+	err := files.Read(paramPath, &fileParams)
+	if err != nil {
+		return err
+	}
+
+	fileParams["parameters"].(map[string]interface{})["params"] = params
+
+	return files.Write(paramPath, fileParams)
+}
+
+func NewInputHandler() InputHandler {
+	return InputHandler{
+		ParamsFile:      ParamsFile,
+		ConnectionsFile: ParamsFile,
+	}
+}
+
+func (i InputHandler) ReadParams(basePath string) ([]byte, error) {
+	paramsPath := path.Join(basePath, i.ParamsFile)
+	fileParams := make(map[string]any)
+
+	err := files.Read(paramsPath, &fileParams)
+
+	if err != nil {
+		return []byte{}, err
+	}
+
+	return json.Marshal(fileParams["parameters"].(map[string]any)["params"].(map[string]any)["value"])
+}
+
+func (i InputHandler) WriteParams(basePath string, params map[string]any) error {
+	paramPath := path.Join(basePath, i.ParamsFile)
```

Comment:
Can these first 5 lines just re-use ReadParams?

Reply:
it could, but then I would have to re serialize the data.